src/core/f-modify.c

```diff
@@ -159,8 +168,16 @@
 			src_ser = Make_Binary(6);
 			src_ser->tail = Encode_UTF8_Char(BIN_HEAD(src_ser), VAL_CHAR(src_val));
 			needs_free = TRUE;
+			limit = -1;
+		}
+		else if (ANY_STR(src_val)) {
+			src_len = VAL_LEN(src_val);
+			if (limit >= 0 && src_len > limit) src_len = limit;
```

Comment:
@hostilefork The `src_len > limit` part may be even unnecessary. I think, callers take care of that.